### PR TITLE
Add missing type definitions for table #8507

### LINKED ISSUE #8507: defaultExpandAllRows missing in typescript TableProps<T>
### Version
3.0.0

### Environment
windows 10

### Reproduction link
[http:///localhost.com](http:///localhost.com)

### Steps to reproduce
```  
return <Table
            columns={this.props.columns}
            dataSource={this.props.data}
            defaultExpandAllRows={true}
            expandedRowRender={(record) => this.expandedRowRender(record.key)}
        />
```

// typescript compiler complains because missing in definition

### What is expected?
Need to be added to TableProps<T> in interface.tsx.

### What is actually happening?
error given by typescript compiler when used

### PR DESCRIPTION
**Pull request by danedavid**
Fixes #8507

### CODE REVIEW

File: components/table/interface.tsx [hunk 1]
```diff
@@ -82,10 +82,12 @@ export interface TableProps<T> {
   rowKey?: string | ((record: T, index: number) => string);
   rowClassName?: (record: T, index: number) => string;
   expandedRowRender?: any;
+  defaultExpandAllRows: boolean;
```
Comment (yesmeck) [inline components/table/interface.tsx]: should be optional

File: components/table/interface.tsx [hunk 2]
```diff
@@ -82,10 +82,12 @@ export interface TableProps<T> {
...
   defaultExpandedRowKeys?: string[] | number[];
   expandedRowKeys?: string[] | number[];
   expandIconAsCell?: boolean;
   expandIconColumnIndex?: number;
+  expandRowByClick: boolean;
```
Comment (yesmeck) [inline components/table/interface.tsx]: same here

**Comment by yesmeck** [comment]
Could you create a another PR? These two props should be optional.

**Comment by danedavid** [comment]
Ok I thought I should do it here. Will raise another

**Pull request merged**